mm/mm_heap/mm_realloc.c

```diff
@@ -87,13 +90,14 @@ FAR void *mm_realloc(FAR struct mm_heap_s *heap, FAR void *oldmem,
     {
       return newmem;
     }
-  else if (size <= CONFIG_MM_HEAP_MEMPOOL_THRESHOLD ||
-           mempool_multiple_alloc_size(heap->mm_mpool, oldmem) >= 0)
+  else if ((blksize =
+            mempool_multiple_alloc_size(heap->mm_mpool, oldmem)) >= 0 ||
+           size <= CONFIG_MM_HEAP_MEMPOOL_THRESHOLD)
     {
       newmem = mm_malloc(heap, size);
       if (newmem != NULL)
         {
-          memcpy(newmem, oldmem, size);
+          memcpy(newmem, oldmem, MIN(size, blksize));
```

Comment:
So what if `blksize >= 0` evaluated to false while `size <= CONFIG_MM_HEAP_MEMPOOL_THRESHOLD` is evaluated to true? This will lead to `memcpy` with a negative size parameter.

Reply:
This situation shows that `oldmem` comes from heap, `newmem` will come from mempool, use `mm_alloc` to get `newmem` from mempool, use `mm_free` to release `oldmem`

Reply:
This does not answer my question. How we are handling possible memcpy with negative casted size?

Reply:
Done.